tests/integration/local/start_api/start_api_integ_base.py

```diff
@@ -58,20 +59,23 @@ def build(cls):
     @classmethod
     def start_api(cls):
         command = "sam"
+        wait_time = 5
         if os.getenv("SAM_CLI_DEV"):
             command = "samdev"
 
         command_list = [command, "local", "start-api", "-t", cls.template, "-p", cls.port]
 
         if cls.container_mode:
             command_list += ["--warm-containers", cls.container_mode]
+            if cls.container_mode == ContainersInitializationMode.EAGER.value:
+                wait_time = 25
```

Comment:
Is this related to `docker pull` time? And if so can we pull the images upfront and validate they are ready locally?

Reply:
yes, mainly the image creation time

Reply:
My worry is, if there will be any network issue, this 25 seconds might not be enough. If we pull and validate their existence in local, that would be safer option.

Reply:
is there some kind of callback we could add here, which periodically polls (with some backoff) the process to check if the containers are created? and we could add an overall timeout as well.